services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/asynch/JobHashProviderImpl.java

```diff
@@ -0,0 +1,60 @@
+package org.sagebionetworks.repo.manager.asynch;
+
+import java.io.UnsupportedEncodingException;
+
+import org.sagebionetworks.repo.model.asynch.AsynchronousRequestBody;
+import org.sagebionetworks.repo.model.dao.table.TableRowTruthDAO;
+import org.sagebionetworks.repo.model.table.DownloadFromTableRequest;
+import org.sagebionetworks.repo.model.table.QueryBundleRequest;
+import org.sagebionetworks.repo.model.table.QueryNextPageToken;
+import org.sagebionetworks.repo.model.table.TableRowChange;
+import org.sagebionetworks.schema.adapter.JSONObjectAdapterException;
+import org.sagebionetworks.schema.adapter.org.json.EntityFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import com.amazonaws.util.Md5Utils;
+
+public class JobHashProviderImpl implements JobHashProvider {
+	
+	@Autowired
+	TableRowTruthDAO tableRowTruthDao;
+
+	@Override
+	public String getJobHash(AsynchronousRequestBody body) {
+		if(body == null){
+			throw new IllegalArgumentException("Body cannot be null");
+		}
+		try {
+			String jsonString = EntityFactory.createJSONStringForEntity(body);
+			return Md5Utils.md5AsBase64(jsonString.getBytes("UTF-8"));

```

Comment:
Include the objectEtag into the hash. This is can have many hits for the same query.